cmd/osmosisd/cmd/root.go

```diff
@@ -700,16 +700,23 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 			cmd.RunE = func(cmd *cobra.Command, args []string) error {
 				serverCtx := server.GetServerContextFromCmd(cmd)
 
-				// overwrite config.toml values
-				err := overwriteConfigTomlValues(serverCtx)
-				if err != nil {
-					return err
-				}
+				// Get flag value for rejecting config defaults
+				rejectConfigDefaults := serverCtx.Viper.GetBool(FlagRejectConfigDefaults)
 
-				// overwrite app.toml values
-				err = overwriteAppTomlValues(serverCtx)
-				if err != nil {
-					return err
+				// overwrite config.toml and app.toml values, if rejectConfigDefaults is false
+				if !rejectConfigDefaults {
+					// Add ctx logger line to indicate that config.toml and app.toml values are being overwritten
+					serverCtx.Logger.Info("Overwriting config.toml and app.toml values with some recommended defaults. To prevent this, set the --reject-config-defaults flag to true.")
+
+					err := overwriteConfigTomlValues(serverCtx)
```

Comment:
AFAIR, there was also a code branch somewhere that would check if the old value is 5 seconds and only then overwrite it. Have we checked/changed that value?

I'm AFK rn but can help with checking this later

Reply:
Whoops yeah, I had that change locally when testing but didn't push, great catch

https://github.com/osmosis-labs/osmosis/pull/7769/commits/4df1520db2e7e847f6cc57e9aeb200e1a754107e

Reply:
Do we have to write ti the file, or can we pass the updated fike to tm? This lgtm!

Maybe worth asking in cometbft issue if we can achieve this w/o writing the file at runtime 

This lgtm for this pr, nice job@

Reply:
I believe this can be done at the sdk level, I will make an issue and we can look into it at another time.